### PR TITLE
fix issue #124 - JUnit expand option

### PR DESCRIPTION
**Pull request by richarda**
@mattwynne Started looking at #124.  For now, this failing unit test highlights the problem.  I'm still looking at [formatter/junit.rb](https://github.com/cucumber/cucumber/blob/master/lib/cucumber/formatter/junit.rb) to see if that's the place to handle `@option[:expand]`

### CODE REVIEW

File: spec/cucumber/formatter/junit_spec.rb [hunk 1]
```diff
@@ -3,193 +3,249 @@
...
+
+            FEATURE
+            # these type of tables shouldn't crash (or generate test cases)
+            it { expect(@doc.to_s).not_to match /milk/ }
+            it { expect(@doc.to_s).not_to match /cookies/ }
+          end
+        end
       end
+      
+      context "In --expand mode" do
```
Comment (richarda) [inline spec/cucumber/formatter/junit_spec.rb]: This is where the new tests start.  Everything above was just moved into a context.

File: spec/cucumber/formatter/junit_spec.rb [hunk 2]
```diff
@@ -4,192 +4,248 @@
...
-
-    class TestDoubleJunitFormatter < Junit
-      attr_reader :written_files
-
-      def write_file(feature_filename, data)
-        @written_files ||= {}
-        @written_files[feature_filename] = data
+module Cucumber
+  module Formatter
+    describe Junit do
```
Comment (richarda) [inline spec/cucumber/formatter/junit_spec.rb]: changed `module Cucumber::Formatter` to
```
module Cucumber
  module Formatter
```
to get `runtime`

**Comment by mattwynne** [comment]
I'm starting to feel guilty about pinging you about this one @richarda - it might well be too horrible to be worth it.

For version 3.0, I want to re-write the formatters to use the new, much simpler formatter API, precisely to get rid of bugs like this. So if it looks too tricky, we can just put it into the 3.0 milestone and expect it to get fixed then.

**Comment by richarda** [comment]
Don't feel guilty @mattwynne ...
I have noticed the reports in our Jenkins builds have Scenario Outlines with missing details.  I'm glad to now know why :smile:

A silver lining:
I do think our team has a tendency to overuse the scenario outline anyway, so if this waits until 3.0, it will be a good reason to take a hard look at improving our features.  I'm sure there are good uses of the outline, but I often see a lot of [incidental details](https://cukes.info/school) in outlines, (redundant examples that don't help clarify behavior), which makes we wonder if the outline is a bit of a crutch for some people (including me).

**Comment by brasmusson** [comment]
The actual fix of the issue is in https://github.com/cucumber/cucumber/commit/75b097507cefc6c5cfdd42ecb1c0392c5fa7c791.

**Pull request merged**